File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnMoodgrid.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* <b>Experimental</b>: GnMoodgrid
*/
public class GnMoodgrid : GnObject {
  private HandleRef swigCPtr;

  internal GnMoodgrid(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnMoodgrid_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnMoodgrid obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnMoodgrid() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnMoodgrid(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnMoodgrid() : this(gnsdk_csharp_marshalPINVOKE.new_GnMoodgrid(), true) {
  }

/**
* Creates a Presentation that represents the type of moodgrid layout to  generate recommendations for. A presentation
* object is the way to access all Mood names and recommendations supported by its layout.
* @param user [in] : valid user 
* @param type [in] : enum value representing the Presentation type . 
* @param coordinate [in] : enum value representing the coordinate type for the presentation layout.
* @return presentation.
*/
  public GnMoodgridPresentation CreatePresentation(GnUser user, GnMoodgridPresentationType type, GnMoodgridCoordinateType coordinate) {
    GnMoodgridPresentation ret = new GnMoodgridPresentation(gnsdk_csharp_marshalPINVOKE.GnMoodgrid_CreatePresentation(swigCPtr, GnUser.getCPtr(user), (int)type, (int)coordinate), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
* Retrieves a data point representing the dimensions of the presentation e.g. 5,5 
* @return datapoint.
*/
  public GnMoodgridDataPoint Dimensions(GnMoodgridPresentationType type) {
    GnMoodgridDataPoint ret = new GnMoodgridDataPoint(gnsdk_csharp_marshalPINVOKE.GnMoodgrid_Dimensions(swigCPtr, (int)type), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
* Version information for the library
* @return version
*/
  public string Version {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnMoodgrid_Version_get(swigCPtr) );
	} 

  }

/** 
* Build Date for the library
* @return build date
*/
  public string BuildDate {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnMoodgrid_BuildDate_get(swigCPtr) );
	} 

  }

  public GnMoodgridProviderEnumerable Providers {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnMoodgrid_Providers_get(swigCPtr);
      GnMoodgridProviderEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnMoodgridProviderEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnMoodgrid.java

package com.gracenote.gnsdk;

/** 
* <b>Experimental</b>: {@link GnMoodgrid} 
*/ 
 
public class GnMoodgrid extends GnObject {
  private long swigCPtr;

  protected GnMoodgrid(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnMoodgrid_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnMoodgrid obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnMoodgrid(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnMoodgrid() {
    this(gnsdk_javaJNI.new_GnMoodgrid(), true);
  }

/** 
* Version information for the library 
* @return version 
*/ 
 
  public static String version() {
    return gnsdk_javaJNI.GnMoodgrid_version();
  }

/**  
* Build Date for the library 
* @return build date 
*/ 
 
  public static String buildDate() {
    return gnsdk_javaJNI.GnMoodgrid_buildDate();
  }

  public GnMoodgridProviderIterable providers() {
    return new GnMoodgridProviderIterable(gnsdk_javaJNI.GnMoodgrid_providers(swigCPtr, this), true);
  }

/** 
* Creates a Presentation that represents the type of moodgrid layout to  generate recommendations for. A presentation 
* object is the way to access all Mood names and recommendations supported by its layout. 
* @param user [in] : valid user  
* @param type [in] : enum value representing the Presentation type .  
* @param coordinate [in] : enum value representing the coordinate type for the presentation layout. 
* @return presentation. 
*/ 
 
  public GnMoodgridPresentation createPresentation(GnUser user, GnMoodgridPresentationType type, GnMoodgridCoordinateType coordinate) throws com.gracenote.gnsdk.GnException {
    return new GnMoodgridPresentation(gnsdk_javaJNI.GnMoodgrid_createPresentation(swigCPtr, this, GnUser.getCPtr(user), user, type.swigValue(), coordinate.swigValue()), true);
  }

/** 
* Retrieves a data point representing the dimensions of the presentation e.g. 5,5  
* @return datapoint. 
*/ 
 
  public GnMoodgridDataPoint dimensions(GnMoodgridPresentationType type) throws com.gracenote.gnsdk.GnException {
    return new GnMoodgridDataPoint(gnsdk_javaJNI.GnMoodgrid_dimensions(swigCPtr, this, type.swigValue()), true);
  }

}
